app/src/main/java/com/example/bookwormadventuresdeluxe2/Book.java

```diff
@@ -68,6 +76,34 @@ public void setStatus(Status status)
     {
         this.status = status;
     }
+
+    /**
+     * Sets the color of an image view based on the given status
+     *
+     * @param status       The status of the item displaying the imageView
+     * @param statusCircle The reference to the imageView to re-color
+     */
+    public void setStatusCircleColor(Status status, ImageView statusCircle)
+    {
+        switch (status)
+        {
+            case Available:
+                statusCircle.getDrawable().setColorFilter(ResourcesCompat.getColor(GlobalApplication.getAppContext().getResources(), R.color.available, null), PorterDuff.Mode.SRC_ATOP);
+                break;
+            case Borrowed:
+                statusCircle.getDrawable().setColorFilter(ResourcesCompat.getColor(GlobalApplication.getAppContext().getResources(), R.color.borrowed, null), PorterDuff.Mode.SRC_ATOP);
+                break;
+            case Requested:
+                statusCircle.getDrawable().setColorFilter(ResourcesCompat.getColor(GlobalApplication.getAppContext().getResources(), R.color.requested, null), PorterDuff.Mode.SRC_ATOP);
+                break;
+            case Accepted:
+                statusCircle.getDrawable().setColorFilter(ResourcesCompat.getColor(GlobalApplication.getAppContext().getResources(), R.color.accepted, null), PorterDuff.Mode.SRC_ATOP);
+                break;
+            default:
+                /* We would not expect any other id */
+                throw new IllegalArgumentException();

```

Comment:
Later we will have more that 4 statuses because of the handoff procedures.

Reply:
Yep. I think those will be stored in the Status enum, this function is to get the color for the UI indicator which I think we agreed will stay at 4.